src/metpy/plots/declarative.py

```diff
@@ -1618,6 +1618,92 @@ def _build(self):
+
+    This trait corresponds to the keyword length in `matplotlib.pyplot.quiver`.
+    """
+
+    arrowkey = Tuple(Float(allow_none=True), Float(allow_none=True), Float(allow_none=True),
+                     Unicode(allow_none=True), Unicode(allow_none=True), default_value=None,
+                     allow_none=True)
+    arrowkey.__doc__ = """Set the characteristics of an arrow key using a tuple of values
+    representing (value, xloc, yloc, position, string).
+
+    Default is `None`.
+
+    If `None`, no vector key will be plotted.
+
+    value default is 100
+    xloc default is 0.85
+    yloc default is 1.02
+    position default is 'E' (options are 'N', 'S', 'E', 'W')
+    label default is an empty string
+
+    If you wish to change a characteristic of the arrowkey you'll need to have a tuple of five
+    elements, fill in the full tuple using `None` for those characteristics you wish to use the
+    default value and put in the new values for the other elements. This trait corresponds to
+    the keyword length in `matplotlib.pyplot.quiverkey`.
+    """
+
+    @observe('arrowscale', 'pivot', 'skip', 'earth_relative', 'color', 'arrowkey')
+    def _set_need_rebuild(self, _):
+        """Handle changes to attributes that need to regenerate everything."""
+        # Because matplotlib doesn't let you just change these properties, we need
+        # to trigger a clear and re-call of quiver()
+        self.clear()
+
+    def _build(self):
+        """Build the plot by calling needed plotting methods as necessary."""
+        x_like, y_like, u, v = self.plotdata
+
+        kwargs = plot_kwargs(u)
+
+        # Conditionally apply the proper transform
+        if 'transform' in kwargs and self.earth_relative:
+            kwargs['transform'] = ccrs.PlateCarree()
+
+        wind_slice = (slice(None, None, self.skip[0]), slice(None, None, self.skip[1]))
+
+        self.handle = self.parent.ax.quiver(
+            x_like[wind_slice], y_like[wind_slice],
+            u.values[wind_slice], v.values[wind_slice],
+            color=self.color, pivot=self.pivot, scale=self.arrowscale, **kwargs)
+
+        if self.arrowkey is not None:
+            if self.arrowkey[0] is None:
+                length = 100
+            else:
+                length = self.arrowkey[0]
+            if self.arrowkey[1] is None:
+                xloc = 0.85
+            else:
+                xloc = self.arrowkey[1]
+            if self.arrowkey[2] is None:
+                yloc = 1.02
+            else:
+                yloc = self.arrowkey[2]
+            if self.arrowkey[3] is None:
+                position = 'E'
+            else:
+                position = self.arrowkey[3]
+            if self.arrowkey[4] is None:
+                label = ''
+            else:
+                label = self.arrowkey[4]
+            self.parent.ax.quiverkey(self.handle, xloc, yloc, length, label=label,
+                                     labelpos=position, labelcolor=self.color)
```

Comment:
I think this is both more succinct and easier to tweak in the future:

```suggestion
            # The order here needs to match the order of the tuple
            key_kwargs = {'U': 100, 'X': 0.85, 'Y': 1.02, 'labelpos': 'E', 'label': ''}
            for name, val in zip(key_kwargs, self.arrowkey):
                if val is not None:
                    key_kwargs[name] = val
            self.parent.ax.quiverkey(self.handle, labelcolor=self.color, **key_kwargs)
```

Technically it also allows the tuple to omit trailing `None`s, though I wouldn't document it. It also allows us to add items to the end without breaking old code.

Reply:
Incidentally, the `for` loop I suggested above can be written:
```python
key_kwargs.update(filter(lambda i: i[1] is not None, zip(key_kwargs, self.arrowkey)))
```
but I'm not at all convinced that's clearer.

Reply:
I like the for loop a bit more for its readability. I've implemented that in the updated PR.